src/main/java/com/example/gsoc/controller/PluginController.java

```diff
@@ -0,0 +1,69 @@
+package com.example.gsoc.controller;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.example.gsoc.model.Plugin;
+import com.example.gsoc.model.UpdateCenter;
+import com.example.gsoc.service.PluginService;
+import com.fasterxml.jackson.core.exc.StreamReadException;
+import com.fasterxml.jackson.databind.DatabindException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+public class PluginController {
+	@Autowired
+	PluginService pluginService;
+	
+	public PluginController() throws StreamReadException, DatabindException, IOException {
+		readUpdateCenter();
```

Comment:
This is a bad idea. For now, ok to move forward. But we will get ride of this in the future.

Reply:
I agree. Also, this is the outdated copy of the files, here's the [latest one](https://github.com/dheerajodha/plugin-health-scoring/blob/feature-populate-database/src/main/java/io/jenkins/pluginhealth/scoring/controller/PluginController.java) where I've eliminated the call to this method.